src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +36,12 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .mapToObj(i -> (i % 2 == 1) ? numbers.get(i) - 1 : numbers.get(i))
+                .filter(num -> num % 2 != 0)
+                .mapToDouble(Integer::doubleValue)
+                .average()
+                .orElseThrow(() -> new NoSuchElementException("No odd numbers found"));
```

Comment:
How i already said, it would be great to move our "check if even" logic into separate method.
Also, here you can instantly map to double, without "double-mapping" 
```suggestion
                .mapToDouble(i -> (i % 2 == 1) ? numbers.get(i) - 1 : numbers.get(i))
                .filter(num -> num % 2 != 0)   //replace condition with your own method
                .average()
                .orElseThrow(() -> new NoSuchElementException("No odd numbers found"));
```
Plus i would recommend you to try `getAsDouble()` method instead of `orElseThrow`